api-service/src/controllers/CreateQueryTemplate/QueryTemplateValidator.ts

```diff
@@ -0,0 +1,58 @@
+import * as _ from "lodash";
+
+const template_required_variables = process.env.template_required_vars ? process.env.template_required_vars.split(",") : ["DATASET", "STARTDATE", "ENDDATE",];
+
+export const validateTemplate = async (req: Request) => {
+    const type: any = _.get(req, "request.query_type");
+    const query = _.get(req, 'request');
+    const templateData: string = JSON.stringify(query);
+    const validTemplate = isValidTemplate(templateData, type);
+    return { validTemplate };
+}
+
+const isValidTemplate = (templateData: string, type: string) => {
+    let validTemplate = false;
+    const requiredVars = requiredVariablesExist(template_required_variables, getTemplateVariables(templateData));
+    if (!requiredVars) return validTemplate;
+    if (type === "json") {
+        let data = _.cloneDeep(stringifyVars(templateData, type));
+        getTemplateVariables(data).map((variable: string) => {
+            const varRegex = new RegExp(`"{{${variable}}}"`, 'ig');
+            data = data.replace(varRegex, `""`);
+            data = data.replace(/"""/g, `"`);
+        });
+        try {
+            JSON.parse(data);
+            validTemplate = true;
+        } catch (err: any) { console.log(err); validTemplate = false; }
```

Comment:
No console logs it should be in logger with proper logging error

Reply:
resolved